vars/sonarExecuteScan.groovy

```diff
@@ -1,231 +1,88 @@
-            .addIfEmpty('projectVersion', script.commonPipelineEnvironment.getArtifactVersion()?.tokenize('.')?.get(0))
-            .addIfEmpty('githubOrg', script.commonPipelineEnvironment.getGithubOrg())
-            .addIfEmpty('githubRepo', script.commonPipelineEnvironment.getGithubRepo())
-            // check mandatory parameters
-            .withMandatoryProperty('githubTokenCredentialsId', null, { config -> config.legacyPRHandling && isPullRequest() })
-            .withMandatoryProperty('githubOrg', null, { isPullRequest() })
-            .withMandatoryProperty('githubRepo', null, { isPullRequest() })
-            .use()
-
-        if(configuration.options instanceof String)
-            configuration.options = [].plus(configuration.options)
-
-        loadCertificates(configuration)
+        def stepParameters = [:].plus(parameters)
 
-        def worker = { config ->
-            try {
-                withSonarQubeEnv(config.instance) {
-                    List envVars = []
-
-                    loadSonarScanner(config)
-
-                    if(fileExists('.certificates/cacerts')){
-                        envVars = envVars.plus("SONAR_SCANNER_OPTS=-Djavax.net.ssl.trustStore=${env.WORKSPACE}/.certificates/cacerts")
-                    }
+        def script = checkScript(this, parameters) ?: this
+        stepParameters.remove('script')
 
-                    if(config.organization) config.options.add("sonar.organization=${config.organization}")
-                    if(config.projectVersion) config.options.add("sonar.projectVersion=${config.projectVersion}")
-                    // prefix options
-                    config.options = config.options.collect { it.startsWith('-D') ? it : "-D${it}" }
-                    withEnv(envVars){
-                        sh "PATH=\$PATH:'${env.WORKSPACE}/.sonar-scanner/bin' sonar-scanner ${config.options.join(' ')}"
-                    }
+        def utils = parameters.juStabUtils ?: new Utils()
+        stepParameters.remove('juStabUtils')
+
+        def jenkinsUtils = parameters.jenkinsUtilsStub ?: new JenkinsUtils()
+        stepParameters.remove('jenkinsUtilsStub')
+
+        new PiperGoUtils(this, utils).unstashPiperBin()
+        utils.unstash('pipelineConfigAndTests')
+        script.commonPipelineEnvironment.writeToDisk(script)
+
+        writeFile(file: ".pipeline/tmp/${METADATA_FILE}", text: libraryResource(METADATA_FILE))
+
+        withEnv([
+            "PIPER_parametersJSON=${groovy.json.JsonOutput.toJson(stepParameters)}",
+        ]) {
+            // get context configuration
+            Map config = readJSON(text: sh(returnStdout: true, script: "./piper getConfig --contextConfig --stepMetadata '.pipeline/tmp/${METADATA_FILE}'"))
+            echo "Config: ${config}"
+
+            Map stepConfig = readJSON(text: sh(returnStdout: true, script: "./piper getConfig --stepMetadata '.pipeline/tmp/${METADATA_FILE}'"))
+            echo "StepConfig: ${stepConfig}"
```

Comment:
This will not use custom defaultConfigs or a custom config file. See my recent changes to piperExecuteBin, already on master. The code isn't easily re-used, though, maybe I should have added the two methods to PiperGoUtils.

Reply:
#1346

Reply:
I think we should rather tackle the default topic properly via the binary instead of fiddling around with parameters.

From my POV it is okay to not adapt the recent change here and rather rely on a proper solution in the binary default resolution to be done soon.

Reply:
Sure, I just wanted to point it out.

Reply:
I will implement the change to support *custom defaults* and *project configuration* for now. But I do not favor the approach to do this is Groovy.